### PR TITLE
Feature/cplp 1210/service market place

### PR DESCRIPTION
**Pull request by nidhigarg-bmw**

### CODE REVIEW

File: cx-portal/src/components/overlays/ServiceRequest/index.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    }
+  }
+
+  return (
+    <>
+      <DialogHeader
+        title={t('content.serviceMarketplace.headline')}
+        intro={''}
+        closeWithIcon={true}
+        onCloseWithIcon={() => dispatch(show(OVERLAYS.NONE, ''))}
```
Comment (oyo) [inline cx-portal/src/components/overlays/ServiceRequest/index.tsx:67]: there is a redux action `closeOverlay`

File: cx-portal/src/components/overlays/ServiceRequest/index.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            setTermsChecked(e.target.checked)
+          }}
+          onFocusVisible={function noRefCheck() {}}
+        />
+      </DialogContent>
+
+      <DialogActions>
+        <Button
+          variant="outlined"
+          onClick={() => dispatch(show(OVERLAYS.NONE))}
```
Comment (oyo) [inline cx-portal/src/components/overlays/ServiceRequest/index.tsx:93]: there is a redux action `closeOverlay`

File: cx-portal/src/components/pages/ServiceMarketplaceDetail/components/MarketplaceHeader/index.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    if (success) {
+      setTimeout(() => {
+        dispatch(setSuccessType(false))
+      }, 5000)
+      return (
+        <Button
+          color="success"
+          className="subscribe-btn"
+          onClick={() =>
+            UserService.hasRole(ROLES.SUBSCRIBE_SERVICE_MARKETPLACE)
```
Comment (oyo) [inline cx-portal/src/components/pages/ServiceMarketplaceDetail/components/MarketplaceHeader/index.tsx:53]: better enable/disable this button if the role is not present
`disabled={!UserService.hasRole(ROLES.SUBSCRIBE_SERVICE_MARKETPLACE)}`

or maybe even better: not render the button at all?

File: cx-portal/src/components/pages/ServiceMarketplaceDetail/components/MarketplaceHeader/index.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          {t('content.appdetail.subscribe')}
+        </Button>
+      )
+    } else {
+      return (
+        <Button
+          color="primary"
+          className="subscribe-btn"
+          onClick={() =>
+            UserService.hasRole(ROLES.SUBSCRIBE_SERVICE_MARKETPLACE)
```
Comment (oyo) [inline cx-portal/src/components/pages/ServiceMarketplaceDetail/components/MarketplaceHeader/index.tsx:67]: enable/disable - see other comment

File: cx-portal/src/components/pages/ServiceMarketplaceDetail/components/MarketplaceSubscription/MarketplaceSubscription.scss [hunk 5]
```diff
@@ -0,0 +1,19 @@
+.marketplace-subscriptions {
```
Comment (oyo) [inline cx-portal/src/components/pages/ServiceMarketplaceDetail/components/MarketplaceSubscription/MarketplaceSubscription.scss:1]: header missing

File: cx-portal/src/components/pages/ServiceMarketplaceDetail/components/MarketplaceProvider/MarketplaceProvider.scss [hunk 6]
```diff
@@ -0,0 +1,7 @@
+.marketplace-provider {
```
Comment (oyo) [inline cx-portal/src/components/pages/ServiceMarketplaceDetail/components/MarketplaceProvider/MarketplaceProvider.scss:1]: header missing

File: cx-portal/src/components/pages/ServiceMarketplaceDetail/components/MarketplaceHeader/MarketplaceHeader.scss [hunk 7]
```diff
@@ -0,0 +1,34 @@
+.marketplace-header {
```
Comment (oyo) [inline cx-portal/src/components/pages/ServiceMarketplaceDetail/components/MarketplaceHeader/MarketplaceHeader.scss:1]: header missing

File: cx-portal/src/components/pages/ServiceMarketplaceDetail/Marketplace.scss [hunk 8]
```diff
@@ -0,0 +1,15 @@
+.marketplace-main {
```
Comment (oyo) [inline cx-portal/src/components/pages/ServiceMarketplaceDetail/Marketplace.scss:1]: header missing

File: cx-portal/src/components/pages/ServiceMarketplace/style.scss [hunk 9]
```diff
@@ -0,0 +1,16 @@
+.container {
```
Comment (oyo) [inline cx-portal/src/components/pages/ServiceMarketplace/style.scss:1]: header missing

File: cx-portal/src/components/pages/ServiceMarketplace/index.tsx [hunk 10]
```diff
@@ -1,11 +1,67 @@
...
+    <div>
+      <div className="banner-heading">
+        <PageHeader
+          title={t('servicemarketplace.title')}
+          hasSubtract={false}
+          headerHeight={250}
+        />
+      </div>
+      <Box
+        sx={{
```
Comment (oyo) [inline cx-portal/src/components/pages/ServiceMarketplace/index.tsx:19]: we should not mix importing a `style.scss` and adding a lot of coded style here - go with one solution, but not both of them.

File: cx-portal/src/components/pages/ServiceMarketplaceDetail/components/MarketplaceSubscription/MarketplaceSubscriptionNames.tsx [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+  index,
+}: {
+  subscription: SubscriptionData
+  index: number
+}) {
+  const { data } = useFetchSubscriptionQuery(subscription.offerSubscriptionId)
+
+  return (
+    <li key={index}>
+      {'#' + index}
```
Comment (oyo) [inline cx-portal/src/components/pages/ServiceMarketplaceDetail/components/MarketplaceSubscription/MarketplaceSubscriptionNames.tsx:38]: I think you can add this number with pure css by customizing an `<ol>` element

File: cx-portal/src/components/pages/ServiceMarketplaceDetail/index.tsx [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+    setShowSuccessAlert(false)
+  }
+
+  return (
+    <main>
+      <Button
+        color="secondary"
+        size="small"
+        onClick={() => navigate('/servicemarketplace')}
+        sx={{ margin: '50px 0px 0px 90px' }}
```
Comment (oyo) [inline cx-portal/src/components/pages/ServiceMarketplaceDetail/index.tsx:53]: again mix of imported and custom style

**Review by oyo — CHANGES_REQUESTED** [review]
css files should also get copyright headers
it's not a good practice to mix importing .scss and added sx/style={{}} props

**Comment by oyo** [comment]
please also update release notes

**Comment by nidhigarg-bmw** [comment]
@oyo I have fixed all the request changes. Can you please review again.

**Review by oyo — DISMISSED** [review]
looks good now :)

**Review by oyo — APPROVED** [review]
now good - resolved merge conflicts for release notes

**Pull request merged**